CMakeLists.txt

```diff
@@ -20,16 +20,15 @@ cmake_minimum_required (VERSION 2.6)
 
 project (Proton C)
 
+# Enable C++ now for examples and bindings subdirectories, but make it optional.
+enable_language(CXX OPTIONAL)
+
```

Comment:
enable_language(CXX OPTIONAL) enables cmake support for C++ if it is available. I had to move it up from proton-c because CMake requires you do this above any subdirs that will use it (now examples and proton-c). The next block just decides whether or not to use C++ for building proton. The C++ binding is default enabled if C++ is available (checks CXX_COMPILER), and can explicitly enabled/disabled with BUILD_CPP like all other bindings.